browser-test/test.endtoend.ts

```diff
@@ -53,34 +53,29 @@ describe('Run tests against gRPC server', () => {
   const client = new EchoClient(opts);
 
   before(async function() {
-    this.timeout(30000);
+    this.timeout(40000);
 
-    //TODO: Make sure server is up and running before starting tests, but
-    //      without using sleep for some arbitrary amount of time
-    await sleep(10000);
     const request = {
       content: 'test',
     };
     const MAX_RETRIES = 20;
     const TIMEOUT = 1000;
 
-    // Code to make sure server is up before starting tests. Commented
-    // out since it doesn't work quite yet
-
-    // for (let retryCount = 0; retryCount < MAX_RETRIES; ++retryCount) {
-    //     console.log('attempt #', retryCount);
-    //     try {
-    //         await client.echo(request);
-    //     }
-    //     catch (err) {
-    //         console.log('Still waiting for server...');
-    //         console.log(err);
-    //         await sleep(TIMEOUT);
-    //         continue;
-    //     }
-    //     console.log('Server is up and running');
-    //     break;
-    // }
+    // Making sure server is up before starting tests.
+    for (let retryCount = 0; retryCount < MAX_RETRIES; ++retryCount) {
```

Comment:
What happens if we hit the `MAX_RETRIES` and it's not ready yet? This `before` will resolve and none of the tests will work. We need to fail in the `before` if we use all of the retries without successfully reaching the server